hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/impl/AlertsEngineImpl.java

```diff
@@ -469,92 +477,122 @@ private void removeTrigger(Trigger trigger) {
         }
     }
 
+    // We allow concurrent threads to make this call in order to process distributed data in parallel. We
+    // use synchronized blocks to protect pendingData.
     @Override
-    public void sendData(Collection<Data> data) {
+    @Lock(LockType.READ)
+    public void sendData(TreeSet<Data> data) {
```

Comment:
Hmm, I did already change the signature of the API, not sure why you see an IDE issue.